sdkproject/Assets/Mapbox/Unity/Map/AbstractMap.cs

```diff
@@ -440,42 +447,91 @@ private void OnEnable()
 			}
 		}
 
+		// TODO: implement IDisposable, instead?
+		protected virtual void OnDestroy()
+		{
+			if (TileProvider != null)
+			{
+				TileProvider.ExtentChanged -= OnMapExtentChanged;
+			}
+			_mapVisualizer.ClearMap();
+			_mapVisualizer.Destroy();
+			if (_previewOptions.isPreviewEnabled == true)
+			{
+				DisableEditorPreview();
+			}
+		}
+
 		protected virtual void Awake()
 		{
-			// Setup a visualizer to get a "Starter" map.
+			MapOnAwakeRoutine();
+
+			if (_previewOptions.isPreviewEnabled == true)
+			{
+				//Disable Editor Preview 
+				_previewOptions.isPreviewEnabled = false;
+			}
+		}
+
+		// Use this for initialization
+		protected virtual void Start()
+		{
+			if (Application.isPlaying)
+			{
+				StartCoroutine("SetupAccess");
+				if (_initializeOnStart)
+				{
+					SetUpMap();
+				}
+			}
+		}
+
+		private void MapOnAwakeRoutine()
+		{
+			// Destroy any ghost game objects. 
 			foreach (Transform tr in transform)
 			{
 				Destroy(tr.gameObject);
 			}
+			//Destroy default tileproviders that might be orphaned due to serialization. 
+			DestroyTileProvider();
 
+			// Setup a visualizer to get a "Starter" map.
 			_mapVisualizer = ScriptableObject.CreateInstance<MapVisualizer>();
 		}
 
-		// Use this for initialization
-		protected virtual void Start()
+
+		private void MapOnStartRoutine()
```

Comment:
How does this differ from the logic in Start()? Looks like we're just omitting StartCoroutine("SetupAccess")? Can this method be called from Start() so we're not repeating code?